### PR TITLE
run import of commandline  args as background job

### LINKED ISSUE #5799: make more responsive DT startup while opening files not in the library
My workflow usually includes pre sorting RAW files into directories, then converting all of those files to jpegs using DT.

Opening multiple new RAW files from a file manager freezes DT for a long time. This can happen if DT is not yet started and with DT already open.

When DT is closed and I open tens (and sometimes - hundreds) of raw files DT starts but is unresponsive for a very long time. It's also impossible to say if anything is happening or if DT is hung. 
A related problem is that xmp generation in this scenario is done in a single thread (while up to 8 background threads is configured in settings)

Opening files from a file manager while DT is running leads to a strange behaviour: mode is switched from lighttable to darkroom and then some processing starts. It's hard to say if DT is working or not as GUI may freeze or do strange things during initial XMP generation.

Usually, working with multiple files is not a problem for DT though: I can easily have a render task running while I edit next batch of photos. So my guess is that this is a problem with process arguments parsing

It would be nice to decouple somehow program startup and XMP generation for files in arguments. May be by creating a separate 'open' task for program to handle later on.

This may somehow be related to #5782 . While splash screen is not an ideal solution for this problem it's still better than looking up last generated XMP file in a directory.

Let me know if I failed to include any important details.

### PR DESCRIPTION
**Pull request by ralfbrown**
When a large number of paths are given on the commandline, it can take a while to import them.  This patch moves the import into a background job when multiple paths are specified, so that the UI can display immediately without having to wait for the imports to complete.

If only a single file is specified on the command line, it is imported synchronously and dt switches to darkroom mode just as before.  Multiple args are assumed to result in multiple files imported, so dt stays in lighttable mode.

If the user quits dt while the import is still in progress, the import completes before dt actually exits (the UI closes immediately).  The user should be warned not to kill the remaining process or try to restart dt until it finishes.

Fixes #5799.

### CODE REVIEW

File: src/control/jobs/film_jobs.c [hunk 1]
```diff
@@ -159,17 +226,19 @@ static int _film_filename_cmp(gchar *a, gchar *b)
...
-  /* first of all gather all images to import */
-  GList *images = NULL;
-  images = _film_recursive_get_files(film->dirname, recursive, &images);
-  if(images == NULL)
+  // first, gather all images to import if not already given
+  if (!images)
   {
-    dt_control_log(_("no supported images were found to be imported"));
-    return;
+    gboolean recursive = dt_conf_get_bool("ui_last/import_recursive");
```
Comment (TurboGit) [inline src/control/jobs/film_jobs.c:234]: const

**Review by TurboGit — CHANGES_REQUESTED** [review]
Minor comment, otherwise looks good.

**Review by TurboGit — APPROVED** [review]
Looks good, thanks !

**Comment by TurboGit** [comment]
Seems like this has introduced a memory corruption. I cannot launch darktable anymore !

```
(gdb) bt
#0  __opendir (name=0x4 <error: Cannot access memory at address 0x4>)
    at ../sysdeps/posix/opendir.c:89
#1  0x00007ffff76d005b in  () at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#2  0x00007ffff76d00b9 in g_dir_open ()
    at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#3  0x00007ffff7cd9390 in _film_recursive_get_files
    (path=path@entry=0x4 <error: Cannot access memory at address 0x4>, recursive=0, result=result@entry=0x7fffd7ff0238)
    at /home/obry/dev/builds/darktable/build/src/src/control/jobs/film_jobs.c:153
#4  0x00007ffff7cd9a81 in _film_import1
    (job=job@entry=0x5555598911e0, film=film@entry=0x0, images=<optimized out>)
    at /home/obry/dev/builds/darktable/build/src/src/control/jobs/film_jobs.c:236
#5  0x00007ffff7cd9c2f in _pathlist_import_run (job=0x5555598911e0)
    at /home/obry/dev/builds/darktable/build/src/src/control/jobs/film_jobs.c:85
#6  0x00007ffff7cd1d69 in dt_control_job_execute (job=job@entry=0x5555598911e0)
    at /home/obry/dev/builds/darktable/build/src/src/control/jobs.c:300
#7  0x00007ffff7cd25c8 in dt_control_run_job (control=0x555555598d80)
    at /home/obry/dev/builds/darktable/build/src/src/control/jobs.c:319
#8  dt_control_work (ptr=<optimized out>)
    at /home/obry/dev/builds/darktable/build/src/src/control/jobs.c:564
#9  0x00007ffff7951ea7 in start_thread (arg=<optimized out>)
    at pthread_create.c:477
#10 0x00007ffff7a68def in clone ()
```
The film->dirname has garbadge in it !

**Comment by TurboGit** [comment]
@ralfbrown : Can you have a look ? TIA.

**Comment by TurboGit** [comment]
Some hints probably to help fixing that... in film_jobs.c:
- line 85
- 
`  _film_import1(job, NULL, params->imagelist); // import the specified images, creating filmrolls as needed`

- line 236

`    images = _film_recursive_get_files(film->dirname, recursive, &images);`

We read "film->..." but NULL was passed above. Now I don't even understand why this did not crashed when I tested. Probably I missed applied the patch :(

**Comment by ralfbrown** [comment]
See #9351 for the fix.

**Pull request merged**